src/utils/ConcreteModelConverter.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+import type { Realm } from "../realm.js";
+import type { BabelNodeSourceLocation } from "babel-types";
+import {
+  AbstractObjectValue,
+  AbstractValue,
+  BooleanValue,
+  ConcreteValue,
+  FunctionValue,
+  NullValue,
+  NumberValue,
+  ObjectValue,
+  StringValue,
+  SymbolValue,
+  UndefinedValue,
+  PrimitiveValue,
+  ArrayValue,
+  ECMAScriptSourceFunctionValue,
+  Value,
+} from "../values/index.js";
+import * as t from "babel-types";
+import invariant from "../invariant.js";
+import type { FunctionBodyAstNode } from "../types.js";
+import { CompilerDiagnostic } from "../errors.js";
+import { EnumerableOwnProperties, Get } from "../methods/index.js";
+import { Create } from "../singletons.js";
+
+function reportCompileError(realm: Realm, message: string, loc: ?BabelNodeSourceLocation) {
+  let error = new CompilerDiagnostic(message, loc, "PP9000", "RecoverableError");
```

Comment:
In general, there should be a 1-1 correspondence between error code and message. Also, this function is so small that it may as well be inlined. You should also pay attention to the return result of handleError.